addons/product/models/product_template.py

```diff
@@ -602,6 +602,23 @@ def _create_variant_ids(self):
                 variants_to_create += current_variants_to_create
                 variants_to_activate += current_variants_to_activate
 
+            else:
+                all_variants_dict = {}
+                for variant in all_variants:
+                    variant_entry = all_variants_dict.get(variant.combination_indices)
+                    # Prevent duplicated variants to be activated
+                    if variant.active or not variant_entry:
+                        all_variants_dict[variant.combination_indices] = variant
```

Comment:
Check how `existing_variants` is constructed above. I think you want to do the same here, by using a `set`.
You might actually want to just extract `existing_variants` above out of its condition and use it here too.

Reply:
I had to slightly modify `existing_variants` to reverse the order of the recordset. Without this it will activate `product_white_duplicate` instead of `product_white`.

Reply:
I'm surprised you need reverse here and the code above does not. `all_variants` was supposedly sorted by `active` to ensure active variants have priority over archived variants of the same combination.

If in your case both are archived (I suppose this is what happens) and you want to pick the oldest, you might want to ensure `all_variants` are sorted by `active (asc), id (desc)`.

Reply:
Indeed it works like this too, I updated the PR